examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,300 @@
+    """
+    return Confirm.ask(
+        "Would you like the LLM to autonomously continue the debate without "
+        "waiting for user input? or ask for your input first time?",
+        default=False,
+    )
+
+
+def run_debate() -> None:
+    """Execute the main debate logic.
+
+    Orchestrates the debate process, including setup, user input, LLM agent
+    interactions, and final feedback. Handles both user-guided and LLM-
+    delegated debates.
+
+    Raises:
+        Exception: If an unexpected error occurs during the debate process.
+    """
+    try:
+        # Get global settings
+        global_settings = get_global_settings(nocache=True)
+        langroid.utils.configuration.set_global(global_settings)
+
+        # Get base LLM configuration
+        agent_config: AgentConfig = get_base_llm_config()
+        system_messages: SystemMessages = load_system_messages(
+            "system_messages.json"
+        )
+        llm_delegate: bool = is_llm_delegate()
+
+        # Select topic and sides
+        selected_topic_tuple: Optional[tuple] = select_debate_topic(system_messages)
+        if not selected_topic_tuple:
+            logger.error("No topic selected. Exiting.")
+            return
+        topic_name, pro_key, con_key = selected_topic_tuple
+        side: str = select_side(topic_name)
+
+        # Prompt for the number of debate turns
+        max_turns: int = int(
+            Prompt.ask(
+                "How many turns should the debate continue for?",
+                default="4",
+            )
+        )
+        # Create agents for pro, con, and feedback
+        pro_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages[pro_key].message
+            )
+        )
+
+        con_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages[con_key].message
+            )
+        )
+
+        feedback_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages["feedback"].message
+            )
+        )
+
+        logger.info("Pro, Con, and feedback agents Created.")
+
+        # Determine user's side
+        if side == "pro":
+            user_agent, ai_agent = pro_agent, con_agent
+            user_side, ai_side = "Pro", "Con"
+        else:
+            user_agent, ai_agent = con_agent, pro_agent
+            user_side, ai_side = "Con", "Pro"
+
+        logger.info(
+            f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+            f"LLM Delegate: {llm_delegate}"
+        )
+
+        is_user_turn: bool = True
+
+        if side == "pro":
+            agent_name = pro_agent
+            pro_agent.clear_history()
+            logger.info(f"\n{side} Agent ({topic_name}):\n")
+            if llm_delegate:
```

Comment:
I think in case of `llm_delegate =False`, you actually don't want the LLM of the agent to say anything at all, and instead just the user enters their side of the argument? So when you initialize the corresponding agent, you can set the llm component to None (an agent need not have an LLM), so its only responders would be `user_response` and `agent_response` (that latter is not used in your example since there are no tools). Alternatively, you can define a subclass of `ChatAgent` and override the `llm_response` method to return None. But it's easier to just set `llm` to None in the agent (or agent config, check which one works).